src/Admin/Pool.php

```diff
@@ -238,17 +238,13 @@ public function hasAdminByClass($class)
     public function getAdminByAdminCode($adminCode)
     {
         $codes = explode('|', $adminCode);
+        $code = array_shift($codes);
 
-        if (false === $codes) {
+        if ('' === $code) {
             return false;
         }
 
-        $admin = $this->getInstance($codes[0]);
```

Comment:
This will change the current behavior:

```php
try {
    $admin = $pool->getAdminByAdminCode('');
} catch (\Exception $e) {
    //...
}
```
Before:
* This would result in `$codes = explode('|', $adminCode);` to return an array with 1 element which would be an empty string on key `0`, which would be passed into `$this->getInstance($codes[0]);` which would most likely throw an `\InvalidArgumentException`.

Now, after your change:
* Now `\InvalidArgumentException` is catched and `$admin` now contains `false`.

I would advise to indeed remove the `false === $codes` (because `explode()` would never return false), but leave handling an empty string in `$code` up to `$this->getInstance($code)`

Reply:
Although what you're pointing is true, I think that behavior was never intended to work in that way. So, I don't know if it must be kept for BC reasons.

Reply:
I think BC could be kept by throwing at the beginning of `getAdminByAdminCode` if `$adminCode` is empty.

Reply:
Done.

Reply:
I don't think it was meant to change the `getInstance` method, but to add a `'' === $code` and throw in the `getAdminByAdminCode` itself

Reply:
Yes exactly. If I was to do this in modern style, I would even write value objects for this, and forbid `''` in the constructor.

Reply:
Moved the exception to be thrown early at `getAdminByAdminCode()`.
BTW, note that if an empty string (or any nonexistent admin id) is present in some child admin, this method still returns `false`, so I think there is an inconsistency. But if you prefer to keep this inconsistency for BC reasons, I think we should at least deprecate the implicit support for that situation.

Reply:
Sure! Please use a deprecation, there are many users of that package, some of which might use that for weird purposes we don't know about.

Reply:
Deprecation added.

Reply:
Thanks for the help. I've squashed and added the changelog to PR description.

Reply:
See the other thread for more nitpicks (sorry, you're almost there I swear!)